apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/Playground/DocumentParams/DatasetParams/InputsMapper.tsx

```diff
@@ -0,0 +1,124 @@
+import { isNumber } from 'lodash-es'
+
+import {
+  Badge,
+  Button,
+  ClientOnly,
+  Icon,
+  ReactStateDispatch,
+  Select,
+  Text,
+  Tooltip,
+} from '@latitude-data/web-ui'
+import {
+  PlaygroundInput,
+  PlaygroundInputs,
+} from '$/hooks/useDocumentParameters'
+
+import { ParamsSource } from '../index'
+import { UseSelectDataset, type DatasetPreview } from './useSelectDataset'
+import { SelectedDatasetRow } from './useSelectedRow'
+
+function getTooltipValue(input: PlaygroundInput) {
+  if (input === undefined || input === null) {
+    return { isEmpty: true, value: 'No value found' }
+  }
+
+  const value = typeof input === 'string' ? input : input.value
+  const isEmpty = value === ''
+  return {
+    isEmpty,
+    value: isEmpty ? 'Empty value' : value,
+  }
+}
+
+export function InputMapper({
+  inputs,
+  mappedInputs,
+  headersOptions,
+  isLoading,
+  onSelectHeader,
+  setSelectedTab,
+}: {
+  inputs: PlaygroundInputs
+  mappedInputs: SelectedDatasetRow['mappedInputs']
+  headersOptions: DatasetPreview['headersOptions']
+  onSelectHeader: UseSelectDataset['onSelectHeader']
+  isLoading: boolean
+  setSelectedTab: ReactStateDispatch<ParamsSource>
+}) {
+  return (
+    <ClientOnly>
```

Comment:
so use client pragma is not enough?

Reply:
The `use client` doesn't mean client only. Means: "This component is interactive". 

But the problem with `use client` components is that they also render on the server. And is here in this weird occasions when the client do not standard things like reading info from local storage that we get the hydratation warning